connection.go

```diff
@@ -542,6 +543,7 @@ func (c *Connection) SendSystemError(id uint32, span Span, err error) error {
 			LogField{"id", id},
 			ErrField(err),
 		).Warn("Couldn't send outbound frame.")
+		c.opts.FramePool.Release(frame)
 		return fmt.Errorf("failed to send error frame, buffer full")
```

Comment:
looking at the cases, it seems like the only time we call `Release(frame)` is when there's an error being returned Consider:
 * using a `defer` + a return error variable that releases frames only on errors (1 release vs 3, little simpler to follow)
 * OR moving the logic into a separate function that is given a frame, and returns whether to release

(the fewer places frames are released, the better)

Reply:
went with `defer` since the logic is fairly easy to follow